File: Patterns/Manuscript/manuscript/code/java/DesignPatterns/src/main/java/chainofresponsibility/HelpDesk.java
package chainofresponsibility;

public class HelpDesk {
    public static void main(String[] args) {
        LevelOneSupport levelOne = new LevelOneSupport();
        LevelTwoSupport levelTwo = new LevelTwoSupport();

        levelOne.setNext(levelTwo);

        SupportTicket ticket = new SupportTicket(1, "Cannot connect to the internet.");
        levelOne.handleRequest(ticket);

        SupportTicket anotherTicket = new SupportTicket(2, "Computer does not start.");
        levelOne.handleRequest(anotherTicket);
    }
}

File: code/csharp/DesignPatterns/chainofresponsibility/HelpDesk.cs
using System;

public class HelpDesk
{
    public static void ChainOfResponsibilityExample()
    {
        LevelOneSupport levelOne = new LevelOneSupport();
        LevelTwoSupport levelTwo = new LevelTwoSupport();

        levelOne.SetNext(levelTwo);

        SupportTicket ticket = new SupportTicket(1, "Cannot connect to the internet.");
        levelOne.HandleRequest(ticket);

        SupportTicket anotherTicket = new SupportTicket(2, "Computer does not start.");
        levelOne.HandleRequest(anotherTicket);
    }
}
